pkg/cloud/ibmcloud/actuators/machine/actuator.go

```diff
@@ -179,8 +180,59 @@ func (ic *IbmCloudClient) Delete(ctx context.Context, cluster *clusterv1.Cluster
 func (ic *IbmCloudClient) Update(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
 	klog.Infof("Updating machine %v for cluster %v.", machine.Name, cluster.Name)
 
-	klog.Infof("TODO: Not yet implemented")
-	return nil
+	// TODO(xunpan): labels/annotations updating
+
+	if util.IsControlPlaneMachine(machine) {
+		klog.Info("TODO: Master inplace update not yet implemented")
+		return nil
+	}
+
+	guest, err := ic.getGuest(machine)
+	if err != nil {
+		return err
+	}
+
+	if guest == nil {
+		return ic.Create(ctx, cluster, machine)
+	}
+
+	machineSpec, err := ibmcloudv1.MachineSpecFromProviderSpec(machine.Spec.ProviderSpec)
+	if err != nil {
+		return err
+	}
+
+	klog.V(2).Infof("Guest: Domain: %v, MaxMemory: %v, OSReferenceCode: %v, LocalDiskFlag: %v, HourlyBillingFlag: %v, Datacenter: %v, StartCpus: %v",
+		machineSpec.Domain, *guest.MaxMemory, *guest.OperatingSystemReferenceCode, *guest.LocalDiskFlag,
+		machineSpec.HourlyBillingFlag, *guest.Datacenter.Name, *guest.StartCpus)
+	klog.V(2).Infof("Machine: Domain: %v, MaxMemory: %v, OSReferenceCode: %v, LocalDiskFlag: %v, HourlyBillingFlag: %v, Datacenter: %v, StartCpus: %v",
+		machineSpec.Domain, machineSpec.MaxMemory, machineSpec.OSReferenceCode, machineSpec.LocalDiskFlag,
+		machineSpec.HourlyBillingFlag, machineSpec.Datacenter, machineSpec.StartCpus)
+
+	if *guest.Domain == machineSpec.Domain &&
+		*guest.MaxMemory == machineSpec.MaxMemory &&
+		(*guest.OperatingSystemReferenceCode == machineSpec.OSReferenceCode ||
+			// TODO(xunpan): we do not know what is the latest, so latest is fine currently
+			strings.HasSuffix(machineSpec.OSReferenceCode, "_LATEST")) &&
+		*guest.LocalDiskFlag == machineSpec.LocalDiskFlag &&
+		*guest.HourlyBillingFlag == machineSpec.HourlyBillingFlag &&
+		*guest.Datacenter.Name == machineSpec.Datacenter &&
+		*guest.StartCpus == machineSpec.StartCpus {
+		// TODO(xunpan)
+		// currently, resource attribute updating triggers recreating a VM instance
+		// any use case to change:
+		//   - ssh key name
+		//   - ssh key name
+		//   - host name
+		return nil
+	}
+
+	klog.Infof("Recreating VM instances for machine %v of cluster %v.", machine.Name, cluster.Name)
+	err = ic.Delete(ctx, cluster, machine)
+	if err != nil {
+		return err
+	}
```

Comment:
just curious, do IBM cloud wait for delete complete then send back notification? 
openstack used to have a bug that after delete, it's still there because of async then create again with similar info lead to issue...

Reply:
Good point. I encountered a Deletion issue during my test. Not sure of it yet. It may cause the `cluterctl` issue I mentioned above.